ext/js/background/backend.js

```diff
@@ -249,6 +258,20 @@ export class Backend {
         chrome.runtime.onInstalled.addListener(this._onInstalled.bind(this));
     }
 
+    /** @type {import('api').PmApiHandler<'connectToDatabaseWorker'>} */
+    async _onPmConnectToDatabaseWorker(_params, ports) {
+        if (ports !== null && ports.length > 0) {
+            await this._dictionaryDatabase.connectToDatabaseWorker(ports[0]);
```

Comment:
Because the API abstraction does not really fit well for what we are doing here. It allows you to specify an API surface where you can have per-handler types for the params. It also allows you to have some extra parameters on every single handler to work around issues like this. So `ports` here is defined as an extra parameter. It is the full array because it tries to match the postMessage API which gives you a list of ports, and the consumer needs to choose the appropriate indexes out of it. If we made it only a single port then every single listener that wanted to do something with multiple ports in the future would not work in the API.